### ISSUE TITLE
Date picker day buttons touch target fails a11y guidelines in landscape mode

### ISSUE THREAD

**Issue by bleroux** [flutter/flutter#170786]
### Steps to reproduce

- Run the code sample
- Tap on the button to open the date picker
- Switch to landscape mode
- Observe the touch target size of days buttons

### Expected results

A11y guidelines are respected.

How to respect them can be discussed, see https://github.com/flutter/flutter/issues/170786#issuecomment-2983356858

### Actual results

A11y guidelines are not respected:

![Screenshot 2025-06-17 at 13 32 02](https://github.com/user-attachments/assets/a38fb7b0-d743-412c-ad70-4e6479b7c935)

This is because the current implementation makes sure that all days are visible (no scrolling).

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MaterialApp(home: MyHomePage()));
}

class MyHomePage extends StatelessWidget {
  const MyHomePage({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: ElevatedButton(
          onPressed: () {
            showDatePicker(
              context: context,
              initialDate: DateTime.now(),
              firstDate: DateTime(2024),
              lastDate: DateTime(2026),
            );
          },
          child: const Text('Open date picker'),
        ),
      ),
    );
  }
}

```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
Flutter (Channel master, 3.33.0-1.0.pre.559, on macOS 14.7.6 23H626 darwin-arm64, locale en-FR)```
```

</details>

**Comment by bleroux** [comment]
Some context:
- I landed a fix for portrait mode in https://github.com/flutter/flutter/pull/169163.
- There is no specification for the date picker in landscape mode.

I have a fix ready which:
- sets the minimum height for each day button to 48px.
- makes it possible to scroll (needed because all day buttons won't fit if their height is set to 48px).
- Screen recording with the fix:

https://github.com/user-attachments/assets/51385f32-2c29-4657-8b2f-cde1c1f6b0dc

While meeting a11y guidelines, I'm not sure this is a solution we want as it makes few days visible without scrolling.

I tried to compare with other implementations:
- Compose does not implement a landscape mode for Date picker, so we can't get inspiration there.
- Google Calendar implements a landscape mode which show all days similarly to the Flutter date picker. The touch targets does not comply to a11y guideline. The motivation for this seems to be that, as soon as TalkBack is activated, the calendar view is automatically replaced by an input view where the user can enter the date:

https://github.com/user-attachments/assets/3c883d3c-f69d-4884-a01f-4395b41e67e3

@dkwingsmt @chunhtai 
For insights on choosing a solution for this a11y issue where we can't rely on a specification.

**Comment by bleroux** [comment]
I don't have access to the internal bugs reports b/377876657 and b/412307873 (see https://github.com/flutter/flutter/issues/158325#issue-2641344432), so I don't know if this was reported for landscape or portrait mode.

**Comment by bleroux** [comment]
@dkwingsmt Gentle ping to get your insights.

**Comment by chunhtai** [comment]
I think we need guideline on this from the material design team. Will ask a question internally and come back this

**Comment by chunhtai** [comment]
https://yaqs.corp.google.com/eng/q/2016001848527290368 internal only question thread

Gemini has pointed out for the material design for time picker documented for constrained viewport, one should use input mode instead of full picker mode. 
https://m3.material.io/components/date-pickers/guidelines

I think the material design team will comment as such, but I will keep the question open to see if material team also agree

**Comment by chunhtai** [comment]
Material team has suggest the same, they should use input mode when in the landscape mode.

The remaining item for this issue is that https://api.flutter.dev/flutter/material/showDatePicker.html

it should not default to calendar mode, but instead defaults based on the screen orientation
